components/brave_today/browser/channels_controller.cc

```diff
@@ -30,25 +30,34 @@ ChannelsController::ChannelsController(
 ChannelsController::~ChannelsController() = default;
 
 Channels ChannelsController::GetChannelsFromPublishers(
-    const std::string& locale,
     const Publishers& publishers,
     PrefService* prefs) {
   Channels channels;
   const auto& channel_subscriptions = prefs->GetDict(prefs::kBraveNewsChannels);
 
   for (const auto& it : publishers) {
-    for (const auto& channel_id : it.second->channels) {
-      // Don't add channels which already exist.
-      if (base::Contains(channels, channel_id))
-        continue;
-
-      auto channel = mojom::Channel::New();
-      channel->channel_name = channel_id;
-      channel->subscribed =
-          channel_subscriptions.FindBoolByDottedPath(locale + "." + channel_id)
-              .value_or(false);
-
-      channels.insert({channel_id, std::move(channel)});
+    for (const auto& locale_info : it.second->locales) {
+      for (const auto& channel_id : locale_info->channels) {
+        if (!channels.contains(channel_id)) {
+          auto channel = mojom::Channel::New();
+          channel->channel_name = channel_id;
+          channels[channel_id] = std::move(channel);
+        }
+
+        auto& channel = channels[channel_id];
+
+        // We already know we're subscribed to this channel in this locale.
+        if (base::Contains(channel->subscribed_locales, locale_info->locale)) {
+          continue;
+        }
+
+        auto subscribed_in_locale =
+            channel_subscriptions
+                .FindBoolByDottedPath(locale_info->locale + "." + channel_id)
```

Comment:
Might be unlikely, but what if the channel name has a dot in it?

Reply:
:+1: updated to not use bydottedpath